src/osd/OSDMap.cc

```diff
@@ -1022,16 +1022,16 @@ int OSDMap::calc_num_osds()
   return num_osd;
 }
 
-void OSDMap::count_full_nearfull_osds(int *full, int *nearfull) const
+void OSDMap::count_full_nearfull_osds(int& full, int& nearfull) const
```

Comment:
@tchaikov I appreciate the conventions followed. But still it's not the best practice though. What if a pointer to be used as an input variable ? Then this convention should be changed as in_ptrname and out_ptrname. 

In my previous projects, we used to follow a comment header for every function that describes the Purpose of the Function, Input Variable, Output Variable etc.
```
/*
Short Description about function:
Input Variables and their description:
Output Variables and their description:
Return Value:
*/
void functionname ()
{

}
```

The advantages of passing a reference (@tchaikov: You would already aware of these):
1. References guarantees that it is valid, but the pointers doesn't guarantee as it can be null or nullptr or another address.
2. References cannot be changed to refer another variable at runtime. This is a good feature to reserve it as output variable.
3. Better readability of the code (most C programmers won't agree this), as using reference is just like using the variable it is equivalenced with. While pointers need to be dereferenced, the programmer should remember always what it is pointing to.
4. References are more natural, if you want to pass a variable to a function. But pointers are useful for implementing data structures, when the location they point to will frequently change.